lightning/src/routing/network_graph.rs

```diff
@@ -0,0 +1,1596 @@
+		self.network_graph.write().unwrap().process_channel_update(msg, &self.secp_ctx)
+	}
+
+	fn get_next_channel_announcements(&self, starting_point: u64, batch_amount: u8) -> Vec<(msgs::ChannelAnnouncement, Option<msgs::ChannelUpdate>, Option<msgs::ChannelUpdate>)> {
+		let network_graph = self.network_graph.read().unwrap();
+		let mut result = Vec::with_capacity(batch_amount as usize);
+		let mut iter = network_graph.get_channels().range(starting_point..);
+		while result.len() < batch_amount as usize {
+			if let Some((_, ref chan)) = iter.next() {
+				if chan.announcement_message.is_some() {
+					result.push((chan.announcement_message.clone().unwrap(),
+						chan.one_to_two.last_update_message.clone(),
+						chan.two_to_one.last_update_message.clone()));
+				} else {
+					// TODO: We may end up sending un-announced channel_updates if we are sending
+					// initial sync data while receiving announce/updates for this channel.
+				}
+			} else {
+				return result;
+			}
+		}
+		result
+	}
+
+	fn get_next_node_announcements(&self, starting_point: Option<&PublicKey>, batch_amount: u8) -> Vec<msgs::NodeAnnouncement> {
+		let network_graph = self.network_graph.read().unwrap();
+		let mut result = Vec::with_capacity(batch_amount as usize);
+		let mut iter = if let Some(pubkey) = starting_point {
+				let mut iter = network_graph.get_nodes().range((*pubkey)..);
+				iter.next();
+				iter
+			} else {
+				network_graph.get_nodes().range(..)
+			};
+		while result.len() < batch_amount as usize {
+			if let Some((_, ref node)) = iter.next() {
+				if node.announcement_message.is_some() {
+					result.push(node.announcement_message.clone().unwrap());
+				}
+			} else {
+				return result;
+			}
+		}
+		result
+	}
+
+	fn should_request_full_sync(&self, _node_id: &PublicKey) -> bool {
+		//TODO: Determine whether to request a full sync based on the network map.
+		const FULL_SYNCS_TO_REQUEST: usize = 5;
+		if self.full_syncs_requested.load(Ordering::Acquire) < FULL_SYNCS_TO_REQUEST {
+			self.full_syncs_requested.fetch_add(1, Ordering::AcqRel);
+			true
+		} else {
+			false
+		}
+	}
+}
+
+
+const SERIALIZATION_VERSION: u8 = 1;
+const MIN_SERIALIZATION_VERSION: u8 = 1;
+
+impl Writeable for NetGraphMsgHandler {
+	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), ::std::io::Error> {
+		writer.write_all(&[SERIALIZATION_VERSION; 1])?;
+		writer.write_all(&[MIN_SERIALIZATION_VERSION; 1])?;
+
+		let network = self.network_graph.read().unwrap();
+		network.write(writer)?;
+		Ok(())
+	}
+}
+
+/// Arguments for the creation of a NetGraphMsgHandler that are not deserialized.
+/// At a high-level, the process for deserializing a NetGraphMsgHandler and resuming normal operation is:
+/// 1) Deserialize the NetGraphMsgHandler by filling in this struct and calling <NetGraphMsgHandler>::read(reaser, args).
+/// 2) Register the new NetGraphMsgHandler with your ChainWatchInterface
```

Comment:
I dont think you need to do this - we only use the chain_monitor for queries, the NetGraphMsgHandler doesn't (currently) have any need for receiving block updates. Maybe in the future we can do that but its not implemented yet.

Reply:
Does it even make sense to serialize NetGraphMsgHandler at all? The only field there is NetworkGraph, which can be serialized on it's own, and then spawn a new MsgHandler when needed.

Reply:
You mean a NetworkGraph?
Not sure why the whole NetGraphMsgHandler would be useful...